File: Clojure/problem025.clj
#! /usr/bin/env clojure
; Problem 25 - Project Euler
; http://projecteuler.net/index.php?section=problems&id=025

(defn fibodigits [maxdigits]
  (loop [a 1 b 1 i 1]
    (let [len (count (str a))]
      ;(println i a len)
      (if (>= len maxdigits)
        (list i a)
        (recur (bigint b) (bigint (+ a b)) (inc i))))))

(println (fibodigits 1000))

File: Ruby/problem025.rb
#! /usr/bin/env ruby
=begin
Problem 25 - Project Euler
http://projecteuler.net/index.php?section=problems&id=25
=end
a, b, i = 1, 1, 1
while a.to_s.size < 1000
    a, b, i = b, a+b, i+1
end
puts i
